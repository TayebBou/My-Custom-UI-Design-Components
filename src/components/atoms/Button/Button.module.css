.button {
  border: none;
  outline: none;
  cursor: pointer;
}

.button:disabled {
  opacity: 0.2;
  cursor: auto;
}
