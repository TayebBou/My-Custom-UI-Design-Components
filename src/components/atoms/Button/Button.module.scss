@import "../../../global_style_settings/_fonts.scss";

.button {
  border: none;
  outline: none;
  cursor: pointer;
  font: {
    family: $textFont;
  }
}

.button:disabled {
  opacity: 0.4;
  cursor: not-allowed;
  transform: none !important;
  transition: none !important;
}
