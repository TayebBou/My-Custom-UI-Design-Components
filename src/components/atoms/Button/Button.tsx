import { FC } from "react";
import styles from './Button.module.css'

export enum ButtonTypes {
  BUTTON= 'button',
  RESET= 'reset',
  SUBMIT= 'submit',
}

type ButtonProps = {
    children: React.ReactNode
    onClick?: () => void
    className?: string
    type?: ButtonTypes
    disabled?: boolean
}

const Button:FC<ButtonProps> = (props) => {

    const { children, type, className, onClick, disabled} = props

    return (
        <button onClick={onClick} className={styles.button + ' ' + className} type={type} disabled={disabled} >
            {children}
        </button>
    )
}

Button.defaultProps = {
    onClick: () => {},
    className: '',
    type: ButtonTypes.BUTTON,
    disabled: false
}

export default Button