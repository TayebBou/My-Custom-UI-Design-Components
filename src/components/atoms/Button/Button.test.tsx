import { screen, render, fireEvent } from "@testing-library/react";
import Button, { ButtonTypes } from "./Button";

const props = {
  onClick: jest.fn(),
  className: "Button__test",
  type: ButtonTypes.BUTTON,
  disabled: true,
};

describe("Button component", () => {
  test("should render correctly with props", () => {
    // Arrange
    const { asFragment } = render(<Button {...props}>Click Me</Button>);
    // Assert
    expect(asFragment()).toMatchSnapshot();
  });

  test("should render correctly without props", () => {
    // Arrange
    const { asFragment } = render(<Button>Default Button</Button>);
    // Assert
    expect(asFragment()).toMatchSnapshot();
  });

  test("when user click on the Button", () => {
    // Arrange
    render(<Button>test</Button>);
    const button = screen.getByRole("button");
    // Act
    button.onclick = jest.fn();
    fireEvent.click(button);
    // Assert
    expect(button.onclick).toHaveBeenCalled();
  });
});
