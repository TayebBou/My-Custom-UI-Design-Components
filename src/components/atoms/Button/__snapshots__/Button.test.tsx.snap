// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render correctly with props 1`] = `
<DocumentFragment>
  <button
    class="button Button__test"
    disabled=""
    type="button"
  >
    Click Me
  </button>
</DocumentFragment>
`;

exports[`Button component should render correctly without props 1`] = `
<DocumentFragment>
  <button
    class="button "
    type="button"
  >
    Default Button
  </button>
</DocumentFragment>
`;
